blocks/core/ff_container/ff_container-form-line/ff_container-form-line.js

```diff
@@ -1,36 +1,72 @@
 'use strict';
 
 var React = require('react');
+var generateClass = require('../../_lib/_ui/class-utils').generateStandardClass;
 
-function generateClass(base, props) {
-    var classNames = [];
-    props = props || {};
-    classNames.push(base);
-    if (!!props.modifier) classNames.push(base + '--' + props.modifier);
-    if (!!props.classes) classNames.push(props.classes);
-    if (!!props.className) classNames.push(props.className);
-    return classNames.join(' ');
+var formFieldDisplayName = 'FormField',
+    formlineClass = 'ff_container-form-line',
+    formlineItemClass = 'ff_container-form-line__item',
+    formlineSectionClass= 'ff_container-form-line__section';
+
+function addClassesToNode(node) {
+    if (!node) return null;
+
+    var nodeClass = generateClass(formlineItemClass, (node.props || {}));
+
+    if (React.isValidElement(node)) {
+        return React.cloneElement(node, { className: nodeClass });
+    } else {
+        return <span className={nodeClass}>{node}</span>;
+    }
+}
+
+function getNodeName(node){
+    return node.displayName || node.type.displayName || node.type.name || node.type;
```

Comment:
That's just React! The constructor can be a simple function, or a class, and the name can live in different places on either of those.